near-contract-standards/src/event.rs

```diff
@@ -0,0 +1,298 @@
+pub struct NftTransferData<'a> {
+    #[serde(borrow)]
+    pub old_owner_id: Cow<'a, str>,
+    #[serde(borrow)]
+    pub new_owner_id: Cow<'a, str>,
+    #[serde(borrow)]
+    pub token_ids: Vec<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub authorized_id: Option<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub memo: Option<Cow<'a, str>>,
+}
+
+impl<'a> NftTransferData<'a> {
+    pub fn new<O, N, T, A, M>(
+        old_owner_id: O,
+        new_owner_id: N,
+        token_ids: Vec<T>,
+        authorized_id: Option<A>,
+        memo: Option<M>,
+    ) -> NftTransferData<'a>
+    where
+        O: Into<Cow<'a, str>>,
+        N: Into<Cow<'a, str>>,
+        T: Into<Cow<'a, str>>,
+        A: Into<Cow<'a, str>>,
+        M: Into<Cow<'a, str>>,
+    {
+        Self {
+            authorized_id: authorized_id.map(|t| t.into()),
+            old_owner_id: old_owner_id.into(),
+            new_owner_id: new_owner_id.into(),
+            token_ids: token_ids.into_iter().map(|s| s.into()).collect(),
+            memo: memo.map(|t| t.into()),
+        }
+    }
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NftBurnData<'a> {
+    #[serde(borrow)]
+    pub owner_id: Cow<'a, str>,
+    #[serde(borrow)]
+    pub token_ids: Vec<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub authorized_id: Option<Cow<'a, str>>,
+    #[serde(borrow)]
+    pub memo: Option<Cow<'a, str>>,
+}
+
+impl<'a> NftBurnData<'a> {
+    pub fn new<O, T, A, M>(
+        owner_id: O,
+        token_ids: Vec<T>,
+        authorized_id: Option<A>,
+        memo: Option<M>,
+    ) -> NftBurnData<'a>
+    where
+        O: Into<Cow<'a, str>>,
+        T: Into<Cow<'a, str>>,
+        A: Into<Cow<'a, str>>,
+        M: Into<Cow<'a, str>>,
+    {
+        Self {
+            owner_id: owner_id.into(),
+            token_ids: token_ids.into_iter().map(|s| s.into()).collect(),
+            authorized_id: authorized_id.map(|t| t.into()),
+            memo: memo.map(|t| t.into()),
+        }
+    }
+}
+
+impl<'a> NearEvent<'a> {
+    pub fn new_171(version: &'static str, event_kind: Nep171EventKind<'a>) -> Self {
+        NearEvent::Nep171(Nep171Event { version, event_kind })
+    }
+
+    pub fn new_171_v1(event_kind: Nep171EventKind<'a>) -> Self {
+        NearEvent::new_171("1.0.0", event_kind)
+    }
+
+    #[must_use = "don't forget to .emit() the event"]
+    pub fn nft_burn(data: Vec<NftBurnData<'a>>) -> Self {
+        NearEvent::new_171_v1(Nep171EventKind::NftBurn(data))
+    }
+
+    #[must_use = "don't forget to .emit() the event"]
+    pub fn nft_transfer(data: Vec<NftTransferData<'a>>) -> Self {
+        NearEvent::new_171_v1(Nep171EventKind::NftTransfer(data))
+    }
+
+    #[must_use = "don't forget to .emit() the event"]
+    pub fn nft_mint(data: Vec<NftMintData<'a>>) -> Self {
+        NearEvent::new_171_v1(Nep171EventKind::NftMint(data))
+    }
+
+    pub(crate) fn to_json_string(&self) -> String {
+        serde_json::to_string(self).unwrap()
```

Comment:
this should never fail, so would be slightly more ideal to just panic through host function to avoid file-specific data being included. Fine to have this done in a following PR though.